### ISSUE TITLE
Allow "unlisted" replies to not show up in home timeline.

### ISSUE THREAD

**Issue by eobet** [mastodon/mastodon#21245]
### Pitch

Many consecutive _replies_ can confuse and clutter follower timelines because the post just appears without context (and in reverse chronological order).

I'm talking about dialogues, discussions and debates which can happen in threads (which have rather poor support, overall currently) which are mostly relevant to the people actively engaging in said thread.

### Motivation

Apparently, lots of people (including me), don't know what "unlisted" does.

If I understood the developers correctly, _only_ thing it does is hide the post from discoverability (public, local hashtag timelines).

But this means that all your followers still see any replies you make to posts, even if they don't see the original post.

In other words, your followers see any reply you make without the relevant context.

This suggestion is to, as a poster, have the option to remove that post from their timelines (but still have anyone see it if they find and click on the original post). This would be accomplished by having the functionality of "unlisted" replies (and only replies, not original posts) extended.

**Comment by trwnh** [comment]
> But this means that all your followers still see any replies you make to posts, even if they don't see the original post.

This will almost never occur in practice, and it has nothing to do with visibility/scope either. It applies just the same to public or followers-only posts. Your Home timeline includes self-replies (which would have context) and replies to people you follow (which usually have context, unless there's someone you don't follow in the thread). And you can always just load the thread and get ancestors/descendants for context.

I'm really not sure what the point of such a change would be, in any case.  And it would be quite dubious to deliver a post to your followers but *not* expect it to show up in their Home timeline.

**Comment by eobet** [comment]
> I'm really not sure what the point of such a change would be, in any case. And it would be quite dubious to deliver a post to your followers but not expect it to show up in their Home timeline.

First off, you never see posts in your home timeline that you don't understand until you see the "replied to X" at the top and then need to click that to get context? Happens all the time to me.

Second, this would specifically _not_ deliver the post to followers! That's the whole point of this suggestion! 😄 It would only deliver the post to the _thread_.

**Comment by trwnh** [comment]
There's no concept of a "thread" really, though, at least not like you're describing. And if you don't deliver to followers, then they'll never see it.

It could be interesting to have a "thread" similar to a forum topic, but that would be really hard to express over federation (which works like delivering emails rather than fetching a post).

It could also be interesting to have a concept of "comments" as different from "replies", but that would require a lot of fundamental changes as well as requiring protocol extensions.

Fundamentally, I think the *real* issue is the "many consecutive replies" and also the "mostly relevant to the people actively engaging in said thread" bit. This issue happens for two reasons:

1) Character limits force posts to be shorter than they otherwise would have been.
2) Current fedi is not a proper forum or discussion platform, but it gets used as one anyway.

You could at least reduce the issue of multiple replies by either increasing the character limit or getting rid of it or otherwise allowing people to post full articles rather than just notes. Perhaps these articles could explicitly have the concept of "comments", wherein you send the reply privately to the author of the article, and your reply then gets redistributed to anyone who follows the article.

There is some work being done that is similar to this with the upcoming support for "Public Groups", which is not an exact fit for this kind of use case, but does open the door for alternative distribution methods.

Specifically, the protocol challenges surrounding this are:

1) You need a way to signal that you should "comment" instead of "reply".
2) You need a way to signal where you should send your "comment".
3) You need a way to signal that your Note is specifically a "comment" on something else.

These are all open questions right now, though...

**Comment by eobet** [comment]
> There's no concept of a "thread" really

No reason there couldn't be, it seems?

https://mastodon.gamedev.place/@antimnguyen/109386641104941517

Anyway, this was just a small idea to reduce clutter/confusion and keep conversations separate for those who want it.

**Comment by trwnh** [comment]
> No reason there couldn't be, it seems?

I'm not entirely familiar with the codebase, but following Conversations instead of just Accounts and Tags *could* be added. It's just that you won't get very far with just this one change alone. If you want to not deliver to followers, you'd need to change the delivery model to be more similar to direct statuses, but only in certain contexts, and discovering those contexts is an unsolved problem for the federation protocol -- I listed 3 points above on what would need to be added. And then there's the UX changes around having to keep track of threads as something different than a "regular" reply.

In short, it's very tricky and difficult to do, let alone get it right.

**Comment by jrasanen** [comment]
> Many consecutive replies can confuse and clutter follower timelines because the post just appears without context (and in reverse chronological order).

Perhaps this is an issue that could be solved by improving the UI in some way. As replies can be disabled for the home timeline:

<img width="397" alt="" src="https://user-images.githubusercontent.com/13303/203738673-2c0b21b1-1648-4c3d-96f6-061983dc7da6.png">

I personally find it nice to see replies on the home timeline, as it allows me to engage in discussions better and improves visibility in a federated network. Though, I agree that the UI could be better. I think the current design is the best of both worlds because it doesn't let the sender decide, but anyone receiving messages.

**Comment by Wisdom-Chicken** [comment]
This post made rounds recently on the feeds and got significant traction: https://oslo.town/@themayor/109795022241038174

Apparently people expect this to already be how unlisted comments work, but it is not.
Thinks its a good suggestion to make it behave this way. Would help with threads.

**Comment by FediVideos** [comment]
I'd definitely appreciate being able to keep unlisted replies out of people's "Home" timelines unless they're the ones being replied to.

I don't want to spam my followers' Home timeline with replies to topics that they have no interest in, so I've ended up having to do replies almost entirely by mentions-only, but this means that people interested in a thread aren't able to see that a question has been answered already.

**Comment by trwnh** [comment]
i think tying this behavior to "unlisted" is the wrong move, since "unlisted" isn't about the distribution model, it's about discoverability in public timelines. you get a post addressed to followers and you follow the person, so the post gets inserted into your home feed. that's what "following" means. that the post is also `cc: Public` doesn't imply the post should somehow be hidden from your feed. you could technically filter out such posts, but this would be kind of weird; it would be far better to instead introduce the concept of a "conversation"/"thread"/"topic"/"context" that can be followed separately to have it show up in your home feed, and then allow people to send posts to this conversation but *not* to their followers, if they choose. this would require tracking the `context` as more than just a signal that two posts are related, and it would also require tracking the intended `audience` separately from who you send the activity to.

**Comment by FediVideos** [comment]
I've just checked, and cannot see any Unlisted replies on Home, but I can see Public replies. 
(The only exceptions are unlisted replies that have been boosted, or which are the original poster replying to themselves to construct a thread.)

I've tried this on mstdn.social and tech.lgbt.

Has the behaviour of Mastodon been tweaked since this issue was opened?

**Comment by eobet** [comment]
I just found out that sometimes it's impossible to see what someone was replying to on Mastodon, making this issue worse!

The reason, if I understand things correctly, is if you're on a small server where nobody on that sever is following anyone on the server the unlisted post in your timelines is replying to, you not only don't get the context to the reply in your timeline, it's even impossible to see the original post as it won't have been federated/cached on your server (which is a bigger issue... that apparently things like Akkoma has solved, but not Mastodon because that seems focused on larger server functionality).

This can perhaps be mitigated on the web since you can open the post in a new window on the original server, but through certain third party apps on a mobile phone this isn't possible.

**Comment by trwnh** [comment]
@eobet posts are fetched/resolved upward, so replies to non-private posts will not do that.

**Comment by eobet** [comment]
Regardless of the technical explanation, it's a poor user experience currently.

**Comment by trwnh** [comment]
it's not anything a user should experience because of "unlisted". the *only* thing unlisted will do is exclude your post from the public timelines (local, federated, hashtag). the behavior you're describing (not being able to see the replied-to post) only occurs when the post is private (usually followers-only and no one follows that user)
